ci/travis_env_common.sh

```diff
@@ -31,6 +31,7 @@ export ARROW_JS_DIR=${TRAVIS_BUILD_DIR}/js
 export ARROW_INTEGRATION_DIR=$TRAVIS_BUILD_DIR/integration
 export ARROW_RUBY_DIR=$TRAVIS_BUILD_DIR/ruby
 export ARROW_RUST_DIR=${TRAVIS_BUILD_DIR}/rust
+export ARROW_R_DIR=${TRAVIS_BUILD_DIR}/r
```

Comment:
Any strong opinions about "R/" vs. "r/" I am OK with either

Reply:
I think lower case for the outer subdirectory is fine, but the R code has to be in an uppercase `R`. So as long as it doesn't look too weird to have `r/R/*`.